### PR TITLE
refactor: Change the order of punction used for headings.

### PR DESCRIPTION
**Pull request by feanil**
Update the punctuation used for headings in this repo so that it matches
the recommendation from the docs.openedx.org repo and what we have in
our reference documentation:

https://docs.openedx.org/en/latest/documentors/references/quick_reference_rst.html#headings

### CODE REVIEW

File: oeps/architectural-decisions/oep-0003-arch-async-tasks.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
-===================================
+###################################
 OEP-3: Asynchronous Task Management
-===================================
+###################################
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0003-arch-async-tasks.rst:3]: Ah, so does putting a row of punctuation both above and below the line introduce a new heading level?

Is this the preference for document titles? If so, should we update the RST guide?

File: oeps/architectural-decisions/oep-0005/decisions/0005-centralized-devstack.rst [hunk 2]
```diff
@@ -1,26 +1,26 @@
-==========================================
+##########################################
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0005/decisions/0005-centralized-devstack.rst:1]:
```suggestion
```
in following what you did with the other ADRs, should this just be a single line of `#`?

File: oeps/architectural-decisions/oep-0026-arch-realtime-events.rst [hunk 3]
```diff
@@ -28,7 +28,7 @@ OEP-26: Real-time Events
    :depth: 3
 
 Context
--------
+*******
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0026-arch-realtime-events.rst:31]: I think you mixed up `****` and `####` throughout this document (consistently, though: it seems to render fine :)

File: oeps/architectural-decisions/oep-0041-arch-async-server-event-messaging.rst [hunk 4]
```diff
@@ -26,9 +26,9 @@ OEP-41: Asynchronous Server Event Message Format
    * - Resolution
      - Accepted
 
---------
+********
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0041-arch-async-server-event-messaging.rst:29]: same comment as the one in OEP-26: I think the `*` and `#` levels are mixed up.

File: oeps/architectural-decisions/oep-0045-arch-ops-and-config.rst [hunk 5]
```diff
@@ -30,7 +30,7 @@ OEP-45: Configuring and Operating Open edX
 
 
 Abstract
-========
+********
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0045-arch-ops-and-config.rst:33]: same comment as OEP-32: heading levels should be bumped up by one

File: oeps/best-practices/oep-0019-bp-developer-documentation.rst [hunk 6]
```diff
@@ -372,7 +372,7 @@ Helpful rST-related tools
 * Other options are listed on stackoverflow_.
 
 IDEs
-^^^^
+~~~~
```
Comment (kdmccormick) [inline oeps/best-practices/oep-0019-bp-developer-documentation.rst:375]: intentional change?

File: oeps/best-practices/oep-0022-bp-django-caches.rst [hunk 7]
```diff
@@ -21,7 +21,7 @@ OEP-22: Caching in Django
 +-----------------+--------------------------------------------------------+
 
 Summary
--------
+*******
```
Comment (kdmccormick) [inline oeps/best-practices/oep-0022-bp-django-caches.rst:24]: headings need to be bumped up one level

File: oeps/best-practices/oep-0034-bp-lint-amnesty.rst [hunk 8]
```diff
@@ -24,7 +24,7 @@ OEP-34: Lint Amnesty
 +-----------------+--------------------------------------------------------+
 
 Context
--------
```
Comment (kdmccormick) [inline oeps/best-practices/oep-0034-bp-lint-amnesty.rst:27]: headings need to be bumped up one level

File: oeps/best-practices/oep-0042-bp-authentication.rst [hunk 9]
```diff
@@ -29,7 +29,7 @@ OEP-42: Authentication
    :depth: 2
 
 Summary
-=======
+*******
```
Comment (kdmccormick) [inline oeps/best-practices/oep-0042-bp-authentication.rst:32]: headings should be bumped up one level

File: oeps/best-practices/oep-0047-bp-semantic-versioning.rst [hunk 10]
```diff
@@ -29,7 +29,7 @@ OEP-47: Semantic Versioning
    :depth: 3
 
 Summary
-=======
+*******
```
Comment (kdmccormick) [inline oeps/best-practices/oep-0047-bp-semantic-versioning.rst:32]: headings should be bumped up one level

File: oeps/processes/oep-templates/adr-based-template.rst [hunk 11]
```diff
@@ -38,40 +38,40 @@ OEP-XXXX: OEP Template
      - <start - target end dates for review>
 
 Context
--------
+*******
```
Comment (kdmccormick) [inline oeps/processes/oep-templates/adr-based-template.rst:41]: headings should be bumped up one level

File: oeps/processes/oep-templates/external-link-template.rst [hunk 12]
```diff
@@ -38,13 +38,13 @@ OEP-XXXX: OEP Template
      - <start - target end dates for review>
 
 Abstract
---------
+********
```
Comment (kdmccormick) [inline oeps/processes/oep-templates/external-link-template.rst:41]: headings should be bumped up one level

File: oeps/architectural-decisions/oep-0005/decisions/0005-centralized-devstack.rst (hunk 2 above)
Comment (robrap) [reply oeps/architectural-decisions/oep-0005/decisions/0005-centralized-devstack.rst:1]: Note: I'm pretty sure a single-line and double-line of the same character (e.g. `#`) are treated as separate header levels. I'd prefer a single-line (in general), but there are many places where we use double, including OEPs like: oeps/architectural-decisions/oep-0005-arch-containerize-devstack.rst. It probably makes sense to have OEPs and ADRs be consistent.

**Comment by feanil** [comment]
@kdmccormick @robrap thanks for the initial review, I went back and cleaned up all the overline headings so they are gone, all the pages should be more consistent and match the heading styles in ADRs as well.

File: oeps/architectural-decisions/oep-0026/caliper-realtime-events.rst [hunk 13]
```diff
@@ -1,29 +1,29 @@
 .. _caliper_realtime_events:
 
-========================
+########################
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0026/caliper-realtime-events.rst:3]: Overline heading

File: oeps/architectural-decisions/oep-0026/xapi-realtime-events.rst [hunk 14]
```diff
@@ -1,15 +1,15 @@
 .. _xapi_realtime_events:
 
-=====================
+#####################
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0026/xapi-realtime-events.rst:3]: Overline heading

File: oeps/processes/oep-templates/adr-based-template.rst [hunk 15]
```diff
@@ -2,9 +2,9 @@
 
 .. Below is the display in the left sidebar on RTD. Please omit leading 0's
 
-======================
+######################
```
Comment (kdmccormick) [inline oeps/processes/oep-templates/adr-based-template.rst:5]:
```suggestion
```

File: oeps/processes/oep-templates/pep-based-template.rst [hunk 16]
```diff
@@ -2,9 +2,9 @@
 
 .. Below is the display in the left sidebar on RTD. Please omit leading 0's
 
-======================
+######################
 OEP-XXXX: OEP Template
-======================
+######################
```
Comment (kdmccormick) [inline oeps/processes/oep-templates/pep-based-template.rst:7]: This template needs fixing

**Review by kdmccormick — CHANGES_REQUESTED** [review]

File: oeps/processes/oep-templates/pep-based-template.rst (hunk 16 above)
Comment (kdmccormick) [reply oeps/processes/oep-templates/pep-based-template.rst:6]: @feanil  ^

File: oeps/architectural-decisions/oep-0026/xapi-realtime-events.rst (hunk 14 above)
Comment (kdmccormick) [reply oeps/architectural-decisions/oep-0026/xapi-realtime-events.rst]: @feanil ^

File: oeps/architectural-decisions/oep-0026/caliper-realtime-events.rst (hunk 13 above)
Comment (kdmccormick) [reply oeps/architectural-decisions/oep-0026/caliper-realtime-events.rst]: @feanil  ^

File: oeps/processes/oep-templates/external-link-template.rst [hunk 17]
```diff
@@ -2,9 +2,9 @@
 
 .. Below is the display in the left sidebar on RTD. Please omit leading 0's
 
-======================
+######################
```
Comment (kdmccormick) [inline oeps/processes/oep-templates/external-link-template.rst:5]: Overline heading

File: oeps/architectural-decisions/oep-0026/xapi-realtime-events.rst [hunk 18]
```diff
@@ -1,15 +1,14 @@
...
 
 .. contents::
    :local:
    :depth: 2
 
 Standardized Integration with xAPI
-==================================
+##################################
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0026/xapi-realtime-events.rst:11]: Headings need to be bumped down a level

File: oeps/architectural-decisions/oep-0026/caliper-realtime-events.rst [hunk 19]
```diff
@@ -1,29 +1,28 @@
...
 
 .. contents::
    :local:
    :depth: 3
 
 Standardized Integration with Caliper
-=====================================
+#####################################
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0026/caliper-realtime-events.rst:11]: Headings need to be bumped down a level

File: oeps/architectural-decisions/oep-0049-django-app-patterns.rst [hunk 20]
```diff
@@ -1,7 +1,7 @@
 
-===========================
+###########################
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0049-django-app-patterns.rst:2]: Overline heading

File: oeps/architectural-decisions/oep-0048-brand-customization.rst [hunk 21]
```diff
@@ -1,7 +1,7 @@
 
-===========================
+###########################
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0048-brand-customization.rst:2]: Overline heading & subheadings need to be bumped down a level

File: oeps/architectural-decisions/oep-0032-arch-unique-identifier-for-users.rst [hunk 22]
```diff
@@ -27,7 +27,7 @@ OEP-32: Unique Identifier for Users
      - 2019-04-15 - 2019-04-29
 
 Decision
-========
+********
```
Comment (kdmccormick) [inline oeps/architectural-decisions/oep-0032-arch-unique-identifier-for-users.rst:30]: Title of this doc uses an overline heading

**Review by kdmccormick — CHANGES_REQUESTED** [review]
Just a few more stragglers

**Review by kdmccormick — APPROVED** [review]
```
Looks good to me
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀
```

**Pull request merged**